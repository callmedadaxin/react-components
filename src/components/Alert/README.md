#### 基本使用

``` js
<div>
  <Alert className="mgb20" message="这是个消息" type="success" description="束带结发了桑德菲杰圣诞节疯狂水电费"/>
  <Alert className="mgb20" message="这是个消息" type="error" description="束带结发了桑德菲杰圣诞节疯狂水电费"/>
</div>
```