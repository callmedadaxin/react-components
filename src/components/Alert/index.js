/*
 * @Author: wangweixin
 * @Date: 2017-11-30 15:11:38
 * @Last Modified by: wangweixin
 * @Last Modified time: 2019-01-17 19:57:54
 */
import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import Icon from "../Icon";
import successIcon from "./images/success.svg";
import errorIcon from "./images/error.svg";

const typeMap = {
  success: successIcon,
  error: errorIcon
};

/**
 * 基本的消息提示
 * 目前用于表单提示后的信息
 * 注： 目前只封装了success, error两种样式，待设计完善
 */
export default class Alert extends PureComponent {
  render() {
    const { message, description, type, className, style } = this.props;
    const classes = classNames("base-alert", `base-alert-${type}`, className);
    return (
      <div data-show="true" className={classes} style={style}>
        {typeMap[type] ? (
          <Icon className="base-alert-icon" link={typeMap[type]} />
        ) : (
          ""
        )}
        <div className="base-alert-content">
          <p className="base-alert-message">{message}</p>
          <p className="base-alert-description">{description}</p>
        </div>
      </div>
    );
  }
}
Alert.defaultProps = {
  type: "success"
};
Alert.propTypes = {
  /** 提示标题 */
  message: PropTypes.string,
  /** 提示描述 */
  description: PropTypes.string,
  /** 提示类型 */
  type: PropTypes.oneOf(["success", "error"])
};
