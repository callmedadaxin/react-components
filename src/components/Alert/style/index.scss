@import "../../../styles/assets/constant.scss";

.base-alert {
  display: flex;
  width: 100%;
  padding: 14px 20px;
  line-height: 14px;

  &.base-alert-error {
    background-color: rgba(238, 70, 48, 0.15);

    .base-alert-icon {
      fill: #ee4630;
    }
  }
  &.base-alert-success {
    background-color: rgba(164, 219, 52, 0.2);

    .base-alert-icon {
      fill: #80d225;
    }
  }

  .base-alert-icon {
    margin-right: 20px;
    width: 30px;
    height: 30px;
  }

  .base-alert-message {
    margin: 0;
    margin-bottom: 6px;
    color: $primaryDarkColor;
    font-weight: bold;
    font-size: 14px;
  }

  .base-alert-description {
    margin: 0;
    font-size: 12px;
    color: $font-color-normal;
  }
}
