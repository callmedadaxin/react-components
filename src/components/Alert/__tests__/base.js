import Alert from '../index'
import React from 'react'
import {shallow} from 'enzyme'

const props = {
  message: '测试',
  description: 'aaa'
}
describe('Alert', () => {
  it('render success alert', () => {
    const comp = shallow(<Alert {...props} type="success"/>)
    expect(comp).toMatchSnapshot()

    expect(comp.find('.base-alert-success').length).toBe(1)
    expect(comp.find('.base-alert-message').text()).toEqual(props.message)
    expect(comp.find('.base-alert-description').text()).toEqual(props.description)
  })
  it('render error alert', () => {
    const comp = shallow(<Alert {...props} type="error"/>)
    expect(comp).toMatchSnapshot()

    expect(comp.find('.base-alert-error').length).toBe(1)
  })
})