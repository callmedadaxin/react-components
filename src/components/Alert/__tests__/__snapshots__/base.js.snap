// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert render error alert 1`] = `
<div
  className="base-alert base-alert-error"
  data-show="true"
>
  <Icon
    className="base-alert-icon"
    link="test-file-stub"
  />
  <div
    className="base-alert-content"
  >
    <p
      className="base-alert-message"
    >
      测试
    </p>
    <p
      className="base-alert-description"
    >
      aaa
    </p>
  </div>
</div>
`;

exports[`Alert render success alert 1`] = `
<div
  className="base-alert base-alert-success"
  data-show="true"
>
  <Icon
    className="base-alert-icon"
    link="test-file-stub"
  />
  <div
    className="base-alert-content"
  >
    <p
      className="base-alert-message"
    >
      测试
    </p>
    <p
      className="base-alert-description"
    >
      aaa
    </p>
  </div>
</div>
`;
